scripts/ownerslint.ts

```diff
@@ -0,0 +1,69 @@
+import chalk from 'chalk';
+import {readdirSync, readFileSync, statSync} from 'fs';
+import {IMinimatch, Minimatch} from 'minimatch';
+import {join} from 'path';
+
+/** Path for the Github owners file. */
+const ownersFilePath = '.github/CODEOWNERS';
+
+/** Path for the .gitignore file. */
+const gitIgnorePath = '.gitignore';
+
+/** Script that lint the CODEOWNERS file and makes sure that all files have an owner. */
+let errors = 0;
+const ownedPaths = readFileSync(ownersFilePath, 'utf8').split('\n')
+    // Trim lines.
+    .map(line => line.trim())
+    // Remove empty lines and comments.
+    .filter(line => line && !line.startsWith('#'))
+    // Split off just the path glob.
+    .map(line => line.split(/\s+/)[0])
+    // Turn paths into Minimatch objects.
+    .map(path => new Minimatch(path, {dot: true, matchBase: true}));
+
+const ignoredPaths = readFileSync(gitIgnorePath, 'utf8').split('\n')
+    // Trim lines.
+    .map(line => line.trim())
+    // Remove empty lines and comments.
+    .filter(line => line && !line.startsWith('#'))
+    // Turn paths into Minimatch objects.
+    .map(path => new Minimatch(path, {dot: true, matchBase: true}));
+
+for (let paths = getChildPaths('.'); paths.length;) {
+  paths = Array.prototype.concat(...paths
+      // Remove ignored paths
+      .filter(path => !ignoredPaths.reduce(
+          (isIgnored, ignoredPath) => isIgnored || ignoredPath.match('/' + path), false))
+      // Remove paths that match an owned path.
+      .filter(path => !ownedPaths.reduce(
+          (isOwned, ownedPath) => isOwned || isOwnedBy(path, ownedPath), false))
+      // Report an error for any files that didn't match any owned paths.
+      .filter(path => {
+        if (statSync(path).isFile()) {
+          console.log(chalk.red(`No code owner found for "${path}".`));
+          errors++;
```

Comment:
Rather than just counting the number of errors that happen, can we enumerate what the files that are missing are?  It shouldn't be too long of a list typically so I think its reasonable to list out what is missing.

Reply:
That's what's happening in the `console.log` above. The log will look something like this:

```
No code owner found for foo.js
No code owner found for bar.js
throw Error('Found 2 files with no owner');
```